src/System.Private.CoreLib/shared/System/IO/Path.cs

```diff
@@ -807,5 +807,23 @@ internal static StringComparison StringComparison
                     StringComparison.OrdinalIgnoreCase;
             }
         }
+
+        /// <summary>
+        /// Trims the ending directory separator if present.
+        /// </summary>
+        /// <param name="path"></param>
+        public static string TrimEndingDirectorySeparator(string path) =>
+            PathInternal.EndsInDirectorySeparator(path.AsSpan()) && !PathInternal.IsRoot(path.AsSpan()) ?
```

Comment:
It would be worth measuring to see if getting path.AsSpan() into a local would make things faster.

Reply:
ok I'll do it.